pyaedt/workflows/templates/PyAEDT_utils.py

```diff
@@ -0,0 +1,143 @@
+        if is_exe(program):
+            return program
+    else:
+        for path in os.environ["PATH"].split(os.pathsep):
+            path = path.strip('"')
+            exe_file = os.path.join(path, program)
+            if is_exe(exe_file):
+                return exe_file
+
+    return None
+
+
+def show_error(msg, oDesktop):
+    oDesktop.AddMessage("", "", 2, str(msg))
+    MessageBox.Show(str(msg), "Error", MessageBoxButtons.OK, MessageBoxIcon.Error)
+    sys.exit()
+
+
+def environment_variables(oDesktop):
+    os.environ["PYAEDT_SCRIPT_PROCESS_ID"] = str(oDesktop.GetProcessID())
+    version = str(oDesktop.GetVersion()[:6])
+    os.environ["PYAEDT_SCRIPT_VERSION"] = version
+    if version > "2023.1":
+        os.environ["PYAEDT_SCRIPT_PORT"] = str(oDesktop.GetGrpcServerPort())
+    else:
+        os.environ["PYAEDT_SCRIPT_PORT"] = str(0)
+    if "Ansys Student" in str(oDesktop.GetExeDir()):
+        os.environ["PYAEDT_STUDENT_VERSION"] = "True"
+    else:
+        os.environ["PYAEDT_STUDENT_VERSION"] = "False"
+    if is_linux:
+        edt_root = os.path.normpath(oDesktop.GetExeDir())
+        os.environ["ANSYSEM_ROOT{}".format(version)] = edt_root
+        ld_library_path_dirs_to_add = [
+            "{}/commonfiles/CPython/3_7/linx64/Release/python/lib".format(edt_root),
+            "{}/commonfiles/CPython/3_10/linx64/Release/python/lib".format(edt_root),
+            "{}/common/mono/Linux64/lib64".format(edt_root),
+            "{}/Delcross".format(edt_root),
+            "{}".format(edt_root),
+        ]
+        os.environ["LD_LIBRARY_PATH"] = ":".join(ld_library_path_dirs_to_add) + ":" + os.getenv("LD_LIBRARY_PATH", "")
+        if version > "2023.1":
+            os.environ["TCL_LIBRARY"] = os.path.join(
+                "{}/commonfiles/CPython/3_10/linx64/Release/python/lib".format(edt_root), "tcl8.5"
+            )
+            os.environ["TK_LIBRARY"] = os.path.join(
+                "{}/commonfiles/CPython/3_10/linx64/Release/python/lib".format(edt_root), "tk8.5"
+            )
+            os.environ["TKPATH"] = os.path.join(
+                "{}/commonfiles/CPython/3_10/linx64/Release/python/lib".format(edt_root), "tk8.5"
+            )
+        else:
+            os.environ["TCL_LIBRARY"] = os.path.join(
+                "{}/commonfiles/CPython/3_7/linx64/Release/python/lib".format(edt_root), "tcl8.5"
+            )
+            os.environ["TK_LIBRARY"] = os.path.join(
+                "{}/commonfiles/CPython/3_7/linx64/Release/python/lib".format(edt_root), "tk8.5"
+            )
+            os.environ["TKPATH"] = os.path.join(
+                "{}/commonfiles/CPython/3_7/linx64/Release/python/lib".format(edt_root), "tk8.5"
+            )
+
+
+def generate_unique_name(root_name, suffix="", n=6):
+    char_set = string.ascii_uppercase + string.digits
+    unique_name = root_name + "_" + "".join(random.choice(char_set) for _ in range(n))  # nosec
+    if suffix:
+        unique_name += suffix
+    return unique_name
```

Comment:
If possible, try to use the method available in pyaedt. This is (at least) the 3rd time I'm seeing such a method in the code base (I'll clean up when I can). Of course, if this is suppose to work without installing pyaedt then I didn't say anything !